src/System.Memory/src/System/Buffers/Text/Base64Encoder.cs

```diff
@@ -40,11 +40,13 @@ public static OperationStatus EncodeToUtf8(ReadOnlySpan<byte> bytes, Span<byte>
             int destIndex = 0;
             int result = 0;
 
-            if (destLength >= GetMaxEncodedToUtf8Length(srcLength))
+            ref byte encodingMap = ref s_encodingMap[0];
+
+            if (srcLength <= MaximumEncodeLength && destLength >= GetMaxEncodedToUtf8Length(srcLength))
```

Comment:
Would it be better to instead cap the `srcLength` to whatever fits here instead of having two copies of the encoding loop? It would avoid the penalty for the case where the output buffer is not big enough.